src/DataProtection/EntityFrameworkCore/src/EntityFrameworkCoreXmlRepository.cs

```diff
@@ -43,7 +43,7 @@ public virtual IReadOnlyCollection<XElement> GetAllElements()
             using (var scope = _services.CreateScope())
             {
                 var context = scope.ServiceProvider.GetRequiredService<TContext>();
-                return context.DataProtectionKeys.AsNoTracking().Select(key => TryParseKeyXml(key.Xml)).ToList().AsReadOnly();
+                return context.DataProtectionKeys.AsNoTracking().Select(key => TryParseKeyXml(key.Xml, _logger)).ToList().AsReadOnly();
```

Comment:
Maybe I don't fully understand the issue, but this looks like the exact same issue because you're capturing `_logger` which will effectively capture `this`.

Reply:
@smitpatel believes it wouldn't be captured. It will become a parameter in the query rather than captured.

Reply:
this.Member generates a parameter. this.Method() would capture this.

Reply:
We still seeing this being captured:

![image](https://user-images.githubusercontent.com/8302101/64389089-0ab43f00-cff6-11e9-882b-63dca33f951c.png)

Reply:
I'm going to put _logger in a local before passing it in just in case.

Reply:
It is in expression tree but not in query cache. EF Core generates parameter by evaluating parts of tree.

Reply:
Are you still okay with this being in a local?

Reply:
That works too. I am fine either way.

Reply:
As a sanity check, When does the expression tree become unrooted?

Reply:
In EF Core query compiler. We evaluate partial subtrees which are not server correlated, generate parameters if they can be changed or inline constant of they cannot be changed. Also it detach actual DbContext from the tree. From that point, it is matched against the query cache.